### ISSUE TITLE
Incorrectly symlinked python leads to wrong sys.path

### ISSUE THREAD

**Issue by msarahan** [ContinuumIO/anaconda-issues#870]
There have been a few mysterious cases lately of incorrect sys.path in conda-build builds:

* https://github.com/conda/conda-build/issues/1075
* https://github.com/conda/conda-build/issues/1073

This seems to happen because although the correct PATH is set, the python in that path is a symlink, and does not set sys.path to what you would expect.

```
drwxrwxr-x. 2 ijstokes ijstokes   4096 Jun 30 21:01 .
drwxrwxr-x. 8 ijstokes ijstokes   4096 Jun 30 20:56 ..
-rwxr-xr-x. 1 ijstokes ijstokes    121 Jun 30 20:56 2to3
lrwxrwxrwx. 1 ijstokes ijstokes     26 Jun 30 21:01 activate -> /opt/anaconda/bin/activate
lrwxrwxrwx. 1 ijstokes ijstokes     23 Jun 30 21:01 conda -> /opt/anaconda/bin/conda
-rwxrwxr-x. 1 ijstokes ijstokes   5146 Jun 30 20:56 c_rehash
lrwxrwxrwx. 1 ijstokes ijstokes     28 Jun 30 21:01 deactivate -> /opt/anaconda/bin/deactivate
-rwxr-xr-x. 1 ijstokes ijstokes    186 Jun 30 20:56 easy_install
lrwxrwxrwx. 1 ijstokes ijstokes     12 Jun 30 20:56 easy_install-2.7 -> easy_install
-rwxr-xr-x. 1 ijstokes ijstokes    119 Jun 30 20:56 idle
-rwxrwxr-x. 3 ijstokes ijstokes 664210 May 31 16:24 openssl
-rwxrwxr-x. 3 ijstokes ijstokes   9958 May 31 16:24 .openssl-libcrypto-fix
-rw-rw-r--. 3 ijstokes ijstokes    213 May 31 16:24 .openssl-post-link.sh
-rwxr-xr-x. 1 ijstokes ijstokes    130 Jun 30 20:56 pip
-rwxr-xr-x. 1 ijstokes ijstokes    104 Jun 30 20:56 pydoc
lrwxrwxrwx. 1 ijstokes ijstokes      9 Jun 30 20:56 python -> python2.7
lrwxrwxrwx. 1 ijstokes ijstokes      9 Jun 30 20:56 python2 -> python2.7
lrwxrwxrwx. 1 ijstokes ijstokes     48 Jun 30 20:56 python2.7 -> /opt/anaconda/pkgs/python-2.7.12-0/bin/python2.7
-rwxr-xr-x. 1 ijstokes ijstokes   1707 Jun 30 20:56 python-config
-rwxr-xr-x. 1 ijstokes ijstokes  18567 Jun 30 20:56 smtpd.py
lrwxrwxrwx. 1 ijstokes ijstokes     46 Jun 30 20:56 sqlite3 -> /opt/anaconda/pkgs/sqlite-3.13.0-0/bin/sqlite3
-rwxr-xr-x. 3 ijstokes ijstokes   7785 Mar 16  2015 tclsh8.5
-rwxr-xr-x. 1 ijstokes ijstokes    144 Jun 30 20:56 wheel
-rwxr-xr-x. 3 ijstokes ijstokes   8448 Mar 16  2015 wish8.5
```

I was not able to reproduce this, though - in CentOS 5 docker image, or in OS X.  Indeed, after I cleaned the package cache on Ian's with ```conda clean -pt``` and reinstalled Python, now things work - python2.7 is an actual file, not a link.

@ilanschnell Was there ever a different version of the Python-2.7.12-0 package?  If so, why wasn't the build number updated?

CC @ijstokes

**Comment by ilanschnell** [comment]
@msarahan I fixed the problem for Python 2.7, but haven't uploaded the new packages yet.  The new Python 3.4.5 and 3.5.2 packages will also contain the fix.  For Python 2.7 the fix is:
```
build:
  no_link:
    - bin/python2.7                   [unix]
```
as opposed to `bin/python`.

**Comment by msarahan** [comment]
I'm very confused then  about why I could not reproduce the problem. Oh well, glad to know that you'll have a fix soon.

**Comment by ilanschnell** [comment]
I can reproduce the problem, and also confirm that the suggested fix works.  I uploaded Python 2.7 Unix packages `python-2.7.12-1.tar.bz2`, which contain the fix.  Can anyone else confirm the new packages work (before we close the issue)?

**Comment by ilanschnell** [comment]
I also applied the fix to Python 3.4.5 and 3.5.2
